### DISCUSSION TITLE
[12.11.2024] Blog 5 - first Component and live demo

### DISCUSSION THREAD

**Discussion by bunki1224** [bunki1224/Softwareengineering-Projekt#37] [Show and tell]
In dieser Woche haben wir intensiv an unserer TripAhead App in React gearbeitet. Ein besonderes Highlight war die Erstellung einer neuen, leicht anpassbaren Komponente für die Timeline-Ansicht. Diese Komponente zeigt Sehenswürdigkeiten wie den Meiji-Schrein an und lässt sich flexibel konfigurieren, sodass sich schnell verschiedene Orte hinzufügen lassen.

Hier ein Beispiel, wie der Code für diese Komponente aussieht:
```jsx

<Activity
  title="Tokyo Skytree"
  address="1 Chome-1-2 Oshiage, Sumida City, Tokyo"
  price={20}
  tags={['Sightseeing']}
  rating={5}
  image="./bilder/Skytree.png"
/>
```
![image](https://github.com/user-attachments/assets/510e0e0e-292a-441b-9601-be8d47f410bc)
Die Komponente ist flexibel gestaltet, sodass sich neue Orte oder Aktivitäten ganz einfach hinzufügen lassen, indem man nur die Attribute anpasst.

Zusätzlich zur Komponente arbeiten wir momentan an der Implementierung einer Drag-and-Drop-Funktion für die Timeline. So können Nutzer Aktivitäten in ihrer Reiseplanung leicht hinzufügen und sortieren. Ein weiterer Fortschritt: Die App wurde auf einem Server veröffentlicht, sodass sie jetzt live getestet und demonstriert werden kann.

>Euer TripAhead-Team (JB)

**Comment by themsgcentre** [comment]
Wir finden den Aufbau und das Design eurer Komponente sehr ansprechend und übersichtlich! Besonders bemerkenswert ist die klare Struktur, die sie für den Nutzer sehr intuitiv macht. Die flexible Konfiguration sorgt für einen modularen Aufbau, wodurch eine solide Grundlage geschaffen wird.

Ein sehr guter Fortschritt!

Viele Grüße,
Projekt TDC (HM)

**Comment by oskarkreutzer** [comment]
Euer Fortschritt mit der TripAhead-App und der Timeline-Komponente ist beeindruckend! Die flexible Konfiguration der Komponente macht es leicht, neue Sehenswürdigkeiten hinzuzufügen. Besonders die geplante Drag-and-Drop-Funktion wird die Benutzerfreundlichkeit enorm verbessern. Es ist großartig, dass die App nun live ist und getestet werden kann – das bringt das Projekt einen großen Schritt weiter. Weiter so!

Viele Grüße
Das GymTrackerX Team (OK)

**Comment by schmid-di** [comment]
Der Blogartikel gibt spannende Einblicke in die Entwicklung der TripAhead App. Besonders beeindruckend ist die flexible Komponente für Sehenswürdigkeiten, die mit minimalem Aufwand neue Aktivitäten wie den Meiji-Schrein anzeigen kann. Die Timeline-Ansicht mit Drag-and-Drop-Option klingt intuitiv und benutzerfreundlich und wird das Planen von Reisen sicher erleichtern. Toll, dass die App nun auch live verfügbar ist und getestet werden kann – ein großer Schritt für das TripAhead-Team!

Euer UMOC-Team
/JM /PL
